### PR TITLE
[AIRFLOW-1115] fix github oauth api URL

### PR DESCRIPTION
**Pull request by renzofrigato**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-1115) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1115
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
This PR fixes the oauth flow when github_enterprise host is "github.com" itself. In this case the API hostname is not github.com but api.github.com (https://developer.github.com/v3/#schema)
For all other cases, the existing behavior is correct (https://developer.github.com/enterprise/2.13/v3/#schema) and it is not affected by this PR.
       

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
There doesn't seem to be any test for the auth backends and it is a small change. However please let me know if I need to add a test.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jgao54** [comment]
Given that this is a github *enterprise* auth backend, it's a bit confusing to have code implicitly convert `github.com` to `https://api.github.com` which is GH, not GHE. Why not just set `host=https://api.github.com` in the config file to use this backend for non-enterprise GH auth for your use case.

**Comment by renzofrigato** [comment]
@jgao54 Github uses `github.com` for the oauth calls and  `api.github.com` for the API calls (like fetching user info). This means that if I set `host=https://api.github.com` as you suggest, the oauth requests would fail.
It may be possible to create a new configuration variable like `api_host` or something similar but it seemed a little overkill just for this use case. Please let me know what you think.

**Comment by jgao54** [comment]
@renzofrigato  understood now. thanks for clarifying

File: airflow/contrib/auth/backends/github_enterprise_auth.py [hunk 1]
```diff
@@ -84,17 +84,18 @@ def __init__(self):
...
-                         leaf.strip('/')])
+        if not self.api_url:
+            self.api_url = (
+                'https://api.github.com' if self.ghe_host == 'github.com'
+                else '/'.join(['https:/',
+                               self.ghe_host,
+                               'api',
+                               get_config_param('api_rev')])
+            )
+        return self.api_url + leaf
```
Comment (jgao54) [inline airflow/contrib/auth/backends/github_enterprise_auth.py:98]: hmm, one more thing,  should there be a '/' between the api_url and the leaf? could you also add unit test for this? thanks!

Comment (rentzso) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: Sure, I will! This method is used AFAIK only in this class and the leaf argument is passed with a '/'.
Do you think I should support also the case in which the '/' is missing?
Thanks for the feedback!

Comment (jgao54) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: Ah sorry, the diff view didn't show that far down. No big deal then.

Comment (jgao54) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: Could you maybe add a note about this in the security.rst file (after the GitHub Enterprise (GHE) Authentication section). This auth back end was originally only used for GHE. Now that you've extended the functionality for github.com, may be worth noting in the doc.

Comment (renzofrigato) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: @jgao54 just added a quick explanation in `security.rst` as you suggested.

Comment (jgao54) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: @renzofrigato  thanks!

**Review by jgao54 — APPROVED** [review]

File: airflow/contrib/auth/backends/github_enterprise_auth.py (hunk 1 above)
Comment (rentzso) [reply airflow/contrib/auth/backends/github_enterprise_auth.py:98]: thank you for the review!

**Pull request closed without merging**